bottlecap/src/bin/bottlecap/main.rs

```diff
@@ -375,6 +375,9 @@ async fn extension_loop_active(
                     request_id, deadline_ms, invoked_function_arn
                 );
                 lambda_enhanced_metrics.increment_invocation_metric();
+                let mut p = invocation_processor.lock().await;
+                p.on_invoke_event(request_id);
```

Comment:
Wonder since this is always invoked first, if we might need to change the constructor for the context buffer, simplifying our logic. As I'm not sure if this would guarantee us to always have this operation to exist before any other hook from the telemetry API, we might need some experiment to check!

Reply:
Had a discussion about this with Jordan, will do in a future PR!